src/python/pants/core/goals/lint.py

```diff
@@ -373,8 +498,50 @@ def key(fs: FieldSet) -> str:
             )
         )
 
-    lint_target_requests = (
-        request_type(batch) for request_type, batch in batch_by_type(lint_target_request_types)
+    def partition_request_get(
+        request_type: type[LintRequest],
+    ) -> Get[Partitions]:
+        partition_request_type: type = getattr(request_type, "PartitionRequest")
+        if partition_request_type in target_partitioners:
+            lint_targets_request_type = cast("type[LintTargetsRequest]", request_type)
+            return Get(
+                Partitions,
+                _LintTargetsPartitionRequest,
+                lint_targets_request_type.PartitionRequest(
+                    tuple(
+                        lint_targets_request_type.field_set_type.create(target)
+                        for target in targets
+                        if lint_targets_request_type.field_set_type.is_applicable(target)
+                    )
+                ),
+            )
+        else:
+            assert partition_request_type in file_partitioners
+            return Get(
+                Partitions,
+                _LintFilesPartitionRequest,
+                cast("type[LintFilesRequest]", request_type).PartitionRequest(specs_paths.files),
+            )
+
+    all_partitions = await MultiGet(
+        partition_request_get(request_type) for request_type in lint_request_types
+    )
+    lint_partitions_by_rt = defaultdict(list)
```

Comment:
`rt` at first sounded like `return_type` to me. I recommend being explicit with the var name